x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -249,6 +249,11 @@ public static boolean canPushToSource(Expression exp, Predicate<FieldAttribute>
                 return canPushToSource(not.field(), hasIdenticalDelegate);
             } else if (exp instanceof UnaryScalarFunction usf) {
                 if (usf instanceof RegexMatch<?> || usf instanceof IsNull || usf instanceof IsNotNull) {
+                    if (usf instanceof IsNotNull) {
+                        if (usf.field() instanceof FieldAttribute fa && fa.dataType() == DataTypes.TEXT) {
+                            return true;
+                        }
+                    }
```

Comment:
This piece of logic could also be placed in `isAttributePushable` and apply for all 3 functions above since RegexMatch should translate to a Wildcard query.

Reply:
The issue this PR is addressing is very specific about what query and what field type is applicable to. And I _think_ this is because of some very specific optimizations on the compute engine side. Thus, why this PR is very targeted. @nik9000 can this optimization be extended to `is null`, `like` and `rlike`?

Reply:
Sure! I filed the original issue because it looked like we never pushed null filtering for text fields but we did for other types. Like, `int IS NOT NULL` was getting pushed to lucene but `text IS NOT NULL` wasn't.

The optimization that makes this less impactful is the one where we test if a field is null before loading it from _source. This applies to all things we load from _source - and we *can* load all kinds of things from _source. But we always load `text` fields from _source so it comes up more for them.

I hadn't intended to make anything special about text fields with my issue - my real intent was to say "text fields should support `IS NOT NULL` push-down too.

And, yeah, they should be able to support any sort of `IS NULL` or implied `IS NOT NULL` from `LIKE` or anything else.

Reply:
@costin this cannot be extended to `regex`. This test fails (among others)

```
org.elasticsearch.xpack.esql.qa.single_node.EsqlClientYamlIT > test {p0=esql/81_text_exact_subfields/filter ignore above} FAILED
    java.lang.AssertionError: Failure at [esql/81_text_exact_subfields:170]: field [values] doesn't have length [1]
    Expected: <1>
         but: was <0>
```

meaning `from test | where text_ignore_above LIKE "*long*"` where `text_ignore_above` is of type `text`. The `like` is using the "exact" variant of the text field. I've went ahead and disabled all the "exact" variants so that I can make the `like` work (meaning, actually using the `text` field as the main field attribute) and this specific query above fails on the ES side with:

```
"type": "query_shard_exception",
                "reason": "failed to create query: Fielddata is disabled on [text_ignore_above] in [test1]. Text fields are not optimised for operations that require per-document field data like aggregations and sorting, so these operations are disabled by default. Please use a keyword field instead. Alternatively, set fielddata=true on [text_ignore_above] in order to load field data by uninverting the inverted index. Note that this can use significant memory.",
```

I'll make the change only for "is null" and "is not null" unless there is something else I'm missing.